### ISSUE TITLE
Importing leads to dead kernel in Jupyter and segmentation fault in ipython

### ISSUE THREAD

**Issue by amichuda** [QuantEcon/QuantEcon.py#500]
Hi,

I'm running quantecon 0.4.5

Whenever I try to import quantecon, my jupyter kernel dies, or the kernel hangs in vscode, or if I import it into an ipython terminal, it has a segmentation fault.

Please advise

**Comment by mmcky** [comment]
hi @lordflaron -- does this happen with any other libraries. Area you able to `import pandas as pd` or `import numpy as np`? I can't seem to replicate your issue on `quantecon=0.4.5` in `ipython` or `jupyter`

**Comment by mmcky** [comment]
hi @lordflaron are you still getting this issue? 
I haven't been able to replicate it using `anaconda` distribution of python. 

Did you change versions of `Numba` by any chance after you installed QuantEcon?

**Comment by lordflaron** [comment]
Hi,

Sorry for the late reply. Let me try again.

I'm not using anaconda, but the the pup installed version.

**Comment by SleepingNewton** [comment]
I am also experiencing this issue, I was able to import the module before but I don't know what's changed since then.

I used "pip check quantecon" to check for unfulfilled dependencies, which led me to downgrade pyqt5 (to 5.12.1) and pyqtwebengine (to 5.12.1). pip check now says there are no bad dependencies but the problem still persists. I've also tried downgrading quantecon to several different versions but that didn't work either. 

I'm using python 3.7.4 with anaconda on fedora 30.

**Comment by mmcky** [comment]
hi @SleepingNewton can you confirm which version of `numba` you are using? 
That dependency needs to be pretty current. 

Thanks for reporting the issue.

**Comment by SleepingNewton** [comment]
Hi @mmcky thanks for getting back to me,

I'm using numba 0.45.1

**Comment by mmcky** [comment]
thanks @SleepingNewton -- I am not sure what is causing this. The latest `CI` run is passing: https://travis-ci.org/QuantEcon/QuantEcon.py using `pip` and the following packages

https://github.com/QuantEcon/QuantEcon.py/blob/09ab55004806629219359f7ae06d200222f815e5/.travis.yml#L44

**Comment by SleepingNewton** [comment]
I don't know if this helps; but I just tried to enter help("quantecon") into a python shell and even that made it crash!

**Comment by mmcky** [comment]
@SleepingNewton @lordflaron I am unable to replicate this in any clean environments. I wish I could advise further on this for you. Are you able to try a clean `anaconda` environment using `conda create`
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html

**Comment by SleepingNewton** [comment]
Hi @mmcky @lordflaron, I created clean environments with python 3.7.5 and 3.7.4. Both failed with "Bus error (core dumped)" on trying to import numba. I even tried the same with python 3.6 and 2.7 but no luck.

Could it be my CPU, kernel or something? Here's some info on my system:
CPU: Intel i5 M430 @ 2.27GHz, 4 cores with integrated graphics
Kernel: 5.3.7-200.fc30.x86_64

If any further information would be useful I'd be happy to provide it.

**Comment by mmcky** [comment]
thanks @SleepingNewton -- can you import `numba` directly?

**Comment by SleepingNewton** [comment]
@mmcky no, I can't import `numba` directly in any of those environments.

**Comment by mmcky** [comment]
Not really sure what is going on here. Is it possible you have isntalled anaconda as `root` user and then trying to run as a local user?

https://stackoverflow.com/questions/51626598/bus-error-core-dumped-due-to-permission-changes

It appears it could also indicate faulty memory (hardware):

https://serverfault.com/questions/871518/apt-get-broken-bus-error-core-dumped

Given `numba` is working on all my machines here -- I suspect this is a local machine issue rather than an issue with `numba` or `anaconda`.

**Comment by oyamad** [comment]
Is this more of an issue with `Numba`? Could this numba/numba#4515 be related?

**Comment by SleepingNewton** [comment]
I've just found that `numba` is importable in a clean environment with `python 2.7.16`.
I've tried using `numba` and `quantcon` on a more up to date machine running fedora 30 and it works fine.
I'm going to try to uninstall anaconda and reinstall it, in case I did `sudo` it.

**Comment by SleepingNewton** [comment]
@lordflaron I removed every library and re-installed them with `pip3 install --user` and numba works now! 
Thanks @mmcky for all your help.
